pkg/controller/etcdproxy/helpers.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcdproxy
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/runtime"
+
+	samplev1alpha1 "github.com/xmudrii/etcdproxy-controller/pkg/apis/etcd/v1alpha1"
+)
+
+// newReplicaSet creates a new Deployment for a EtcdStorage resource. It also sets
+// the appropriate OwnerReferences on the resource so handleObject can discover
+// the EtcdStorage resource that 'owns' it.
+func (c *EtcdProxyController) newReplicaSet(etcdstorage *samplev1alpha1.EtcdStorage, replicas int,
+	etcdControllerNamespace, etcdCoreUrl, etcdProxyNamespace, etcdProxyImage string) *appsv1.ReplicaSet {
```

Comment:
why etcdProxyImage as param and as field (look below, it's used via `c`) ?

Reply:
I would either use all status from `c` or all via params and then not as method, but normal function.